File: php/huifier_test.php
<?php

require_once 'huifier.php';

use PHPUnit\Framework\TestCase;

class HuifyTest extends TestCase
{
    public function testRandomWords()
    {
        $this->assertSame('Хуилка', huify('Вилка'));
        $this->assertSame('Хуелик', huify('Толик'));
        $this->assertSame('Хуяблоко', huify('Яблоко'));
        $this->assertSame('Хуясный', huify('Красный'));
        $this->assertSame('Хуюассан', huify('Круассан'));
        $this->assertSame('Хую', huify('Хрю'));
    }

    public function testVowelRules()
    {
        $this->assertSame('Хуярка', huify('Арка'));
        $this->assertSame('Хуедут', huify('Едут'));
        $this->assertSame('Хуёжик', huify('Ёжик'));
        $this->assertSame('Хуиглу', huify('Иглу'));
        $this->assertSame('Хуекна', huify('Окна'));
        $this->assertSame('Хуюрна', huify('Урна'));
        $this->assertSame('Хуик', huify('Стык'));
        $this->assertSame('Хуепос', huify('Эпос'));
        $this->assertSame('Хуюрта', huify('Юрта'));
        $this->assertSame('Хуящик', huify('Ящик'));
    }

    public function testLetterCase()
    {
        $this->assertSame('Хуилка', huify('Вилка'));
        $this->assertSame('Хуилка', huify('вилка'));
        $this->assertSame('Хуилка', huify('вилка'));
        $this->assertSame('Хуилка', huify('вИлКа'));
    }

    public function testNormalization()
    {
        $this->assertSame('Хуй', huify(''));
        $this->assertSame('Хуй', huify(' '));
        $this->assertSame('Хуй', huify('   '));
        $this->assertSame('Хуилка', huify('  Вилка  '));
        $this->assertSame('Хуй', huify('Й'));
        $this->assertSame('Хуя', huify('А'));
    }
}

File: python/huifier_test.py
import unittest
from huifier import huify

class TestHuifyMethod(unittest.TestCase):

    def test_random_words(self):
        self.assertEqual(huify('Вилка'), 'Хуилка')
        self.assertEqual(huify('Толик'), 'Хуелик')
        self.assertEqual(huify('Яблоко'), 'Хуяблоко')
        self.assertEqual(huify('Красный'), 'Хуясный')
        self.assertEqual(huify('Круассан'), 'Хуюассан')
        self.assertEqual(huify('Хрю'), 'Хую')

    def test_vowel_rules(self):
        self.assertEqual(huify('Арка'), 'Хуярка')
        self.assertEqual(huify('Едут'), 'Хуедут')
        self.assertEqual(huify('Ёжик'), 'Хуёжик')
        self.assertEqual(huify('Иглу'), 'Хуиглу')
        self.assertEqual(huify('Окна'), 'Хуекна')
        self.assertEqual(huify('Урна'), 'Хуюрна')
        self.assertEqual(huify('Стык'), 'Хуик')
        self.assertEqual(huify('Эпос'), 'Хуепос')
        self.assertEqual(huify('Юрта'), 'Хуюрта')
        self.assertEqual(huify('Ящик'), 'Хуящик')

    def test_letter_case(self):
        self.assertEqual(huify('Вилка'), 'Хуилка')
        self.assertEqual(huify('вилка'), 'Хуилка')
        self.assertEqual(huify('ВИЛКА'), 'Хуилка')
        self.assertEqual(huify('вИлКа'), 'Хуилка')

    def test_normalization(self):
        self.assertEqual(huify(''), 'Хуй')
        self.assertEqual(huify(' '), 'Хуй')
        self.assertEqual(huify('   '), 'Хуй')
        self.assertEqual(huify('  Вилка  '), 'Хуилка')
        self.assertEqual(huify('Й'), 'Хуй')
        self.assertEqual(huify('А'), 'Хуя')

if __name__ == '__main__':
    unittest.main()
